astropy/coordinates/tests/test_sky_coord.py

```diff
@@ -1382,3 +1382,22 @@ def test_extra_attributes():
     # Finally, check that we can delete such attributes.
     del sc3.obstime
     assert sc3.obstime is None
+
+
+def test_move_to():
+    t1 = Time('2015-01-01T00:00')
+    t2 = Time('2025-01-01T00:00')
+
+    # Check a very simple case first:
+    frame = ICRS(ra=10.*u.deg, dec=0*u.deg,
+                 distance=10.*u.pc,
+                 pm_ra_cosdec=0.1*u.deg/u.yr,
+                 pm_dec=0*u.mas/u.yr,
+                 radial_velocity=0*u.km/u.s)
+    c1 = SkyCoord(frame, obstime=t1)
+    c2 = c1.move_to(new_obstime=t2)
+    c3 = c1.move_to(dt=10*u.year)
+
+    # TODO: we need to actually check something here...remember trigonometry,
+    # though - the source moves on a straight line, so the above won't move by
+    # 1 deg
```

Comment:
My usual approach is to pick a nice total angle, like 30, 45, or 60 degrees...

Reply:
Note also that a lot of the addition stuff is already checked for the representations.

Reply:
As mentioned above, one way to test is to compare to the output of ```starpm```.  The positional difference for the example above is tiny (0.1 mas). The RV difference is quite significant (0.2 km/s)!